### DISCUSSION TITLE
Chat component returning KeyError: "Element ipyvuetify.TextField(label = 'Type a message...'

### DISCUSSION THREAD

**Discussion by francesfeng** [widgetti/solara#464] [General]
Hi, 

I tried the last 2 code examples provided here [https://solara.dev/api/chat](https://solara.dev/api/chat), it runs fine as a stand alone .py file, but shows the following error if export as a component, and used in a package. 

`File "./site-packages/reacton/core.py", line 743, in get_widget
    raise KeyError(f"Element {el} was found to be in a previous render, you may have used a stale element") KeyError: "Element ipyvuetify.TextField(label = 'Type a message...', v_model = '', on_v_model = <function ...11e7364c0>, rounded = True, filled = True, hide_details = True, style_ = 'flex-grow: 1;', disabled = False) was found to be in a previous render, you may have used a stale element"`

I checked the source code quickly, think a possible reason is that ChatInput component doesn't return a value. [source code here](https://github.com/widgetti/solara/blob/0d95c9dbc5f070eba4a72417bedba79ccfc55a71/solara/lab/components/chat.py#L45)

It solves the problem by returning `message_input`, but it would then lose the button component. Would be good to hear how you'd address this issue. 

```
    with solara.Row(style=style_flat, classes=classes):
        def send(*ignore_args):
            if message != "" and send_callback is not None:
                send_callback(message)
                set_message("")

        message_input = solara.v.TextField(
            label="Type a message...",
            v_model=message,
            on_v_model=set_message,
            rounded=True,
            filled=True,
            hide_details=True,
            style_="flex-grow: 1;",
            disabled=disabled,
        )

        use_change(message_input, send, update_events=["keyup.enter"])

        button = solara.v.Btn(color="primary", icon=True, children=[solara.v.Icon(children=["mdi-send"])], disabled=message == "")

        use_change(button, send, update_events=["click"])
```

Btw, also in the last 2 code examples on the website, I think it should use 
`messages = solara.Reactive([])` instead of `messages = solara.use_reactive([])`?

**Comment by iisakkirotko** [comment]
Hi @francesfeng!

You're right that `ChatInput` doesn't return a value. Solara has a built-in method to catch any rendered components when no explicit return value is given. 

I tried quickly to replicate what you described, but exporting and importing the example as a component and worked fine for me. Could you maybe share some of the related code, so I can take a look.

> Btw, also in the last 2 code examples on the website, I think it should use messages = solara.Reactive([]) instead of messages = solara.use_reactive([])?

Indeed, there it should use `solara.Reactive` (or preferably `solara.reactive`), good catch! Feel free to open a PR for this.

**Comment by iisakkirotko** [comment]
To follow up, I managed to accidentally replicate your issue while working on something unrelated today. In my case the issue was fixed by updating reacton.

**Reply by francesfeng** [reply]
HI  @iisakkirotko Thanks for looking into the problem. Sorry that I was offline for a few days, and just saw your message. Glad that it resolved. I find a shortcut in TextField that you can add an icon/button directly, without using a Button to send and another call back 

```
message_input = rv.TextField(
            label="Type a message...",
            v_model=message,
            on_v_model=set_message,
            rounded=True,
            filled=True,
            hide_details=True,
            style_="flex-grow: 1;",
            append_icon="mdi-send",  # use to append a SEND button
            disabled=disabled,
        )

    rv.use_event(message_input, "keydown.enter", send)

    return message_input
```

**Comment by maartenbreddels** [comment]
I fixed the use_reactive/reactive mistake, thank you for reporting that @francesfeng !
I'm assuming the original problem is resolved with updating reacton, if not, let us know.
